src/h264_subscriber.cpp

```diff
@@ -47,7 +48,8 @@ void H264Subscriber::internalCallback(const h264_image_transport_msgs::H264Packe
         ROS_ERROR("Could not decode H264 frame");
         return;
     }
-    convert_ctx = sws_getCachedContext(convert_ctx, picture->width, picture->height, AV_PIX_FMT_YUV420P, picture->width, picture->height, AV_PIX_FMT_BGR24, SWS_FAST_BILINEAR, NULL, NULL, NULL);
+    convert_ctx = sws_getCachedContext(convert_ctx, picture->width, picture->height, AV_PIX_FMT_YUV420P, picture->width, picture->height, AV_PIX_FMT_BGR24, SWS_FAST_BILINEAR,
+                                       nullptr, nullptr, NULL);
     sensor_msgs::ImagePtr out = boost::make_shared<sensor_msgs::Image>();
     out->data.resize(picture->width * picture->height * 3);
     int stride = 3*picture->width;
```

Comment:
One line below - isn't `uint8_t *dst = &out->data[0]` equivalent to `uint8_t *dst = out->data`? In that case you could skip creating `dst` entirely and use `put->data` directly in place of it.

Reply:
That parameter must be of type `uint8_t *const dst[]`, so pointer of pointer.
What we would need to do then is something like this : `&out->data.data()`. Alas it's not possible because `out->data.data()` yields a rvalue of type `unsigned char *`, and you cannot get a reference of that. You are condemned to store that value somewhere or find a better day to do things :laughing: